ouroboros-consensus/docs/report/chapters/storage/immutabledb.tex

```diff
@@ -922,3 +922,22 @@ \subsection{Recovery}
 operations. For example, while appending a block, we let the second disk write
 fail. This is another way of testing whether we can correctly recover from a
 write that was aborted half-way through.
+
+\subsection{Resources}
+\label{immutable:implementation:resources}
+
+Using the \lstinline!ResourceRegistry! introduced in
+\ref{storage:resourceregistry}, the Immutable database has to keep track of file
+handles for the chunk, primary and secondary indices. Such handles will be
+allocated in the resource registry provided and will be closed when closing the
+Immutable database. Similarly, the current implementation spawns a child thread
+which is responsible to expire chunks that have been unused for some time. Such
+thread will also live in the resource registry provided.
+
+Note that in order to preserve the order in the sequence of deallocations, file
+handles cannot be deallocated and re-allocated on demand as this will make them
+be deallocated earlier than resources that hold a reference to
+them. Specifically threads that are spawned by the chain database itself as
```

Comment:
Can we expand on this in a couple of sentences? Maybe by saying something like: 

> If the thread that uses the chunk, primary, and secondary indices would re-allocate a resource, then these will be younger than the registry that contains this thread, and therefore they will be deallocated first. In this case, the thread might attempt to use the closed file handles, since the resource associated with them would be deallocated.

Reply:
Reworded more or less like your suggestion